gAPI.py

```diff
@@ -10,43 +10,51 @@
 from googleapiclient.discovery import build
 
 
-def Create_Service(client_secret_file, api_name, api_version, *scopes):
+def Create_Service(client_secret_file, api_name, api_version, *scopes, token_path='token.pickle'):
+    """
+    Creates a Google API Client.
+
+    Args:
+        client_secret_file (str): path to the credentials.json file
+        api_name (str): the name of the API to connect to (ex: "classroom")
+        api_version (str): the version of the API to connect to (ex: "v1")
+        *scopes (str[]): a list of permission scopes
+        token_path (str): the path to the token.pickle file to use
+
+    Returns:
+        A Resource object with methods for interacting with the service.
+
+    """
     print(client_secret_file, api_name, api_version, scopes, sep='-')
-    CLIENT_SECRET_FILE = client_secret_file
-    API_SERVICE_NAME = api_name
-    API_VERSION = api_version
-    SCOPES = [scope for scope in scopes[0]]
     os.environ['OAUTHLIB_RELAX_TOKEN_SCOPE'] = '1'
     cred = None
 
-    if os.path.exists('token.pickle'):
-        with open('token.pickle', 'rb') as token:
+    # If we already have a token, use that instead of making the user log in
+    if os.path.exists(token_path):
+        with open(token_path, 'rb') as token:
```

Comment:
This is the only functional change here - allow the caller to pass in their own token path. This allows us to run this with a bunch of tokens that might not all be the root `token.pickle` file (for aggregation).

The other changes are just using the parameters instead of storing them in extra variables for readability.